include/deal.II/dofs/dof_accessor.templates.h

```diff
@@ -3075,8 +3075,10 @@ namespace internal
         (void)i;
         // ::DoFHandler only supports a single active fe with index
         // zero
+#ifdef DEBUG
         typedef dealii::DoFAccessor<dim,DoFHandler<dim,spacedim>, level_dof_access> BaseClass;
         Assert (i == 0, typename BaseClass::ExcInvalidObject());
+#endif
```

Comment:
I'm not particularly fond of putting too many `#ifdef` into our code base. Can we just remove the typedef and put the equivalent of `BaseClass` into the Assertion? You'd have to write it as
```
  Assert (i == 0, 
              (typename dealii::DoFAccessor<dim,DoFHandler<dim,spacedim>, level_dof_access>::ExcInvalidObject()));
```
(note the extra parentheses around the exception object) but I still think that's prettier than adding two lines of `#ifdef`s.